src/main/java/com/mathworks/ci/helper/MatlabCommandRunner.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2022 The MathWorks, Inc.
+ */
+
+package com.mathworks.ci.helper;
+
+import com.atlassian.bamboo.build.logger.BuildLogger;
+import com.atlassian.bamboo.process.ExternalProcessBuilder;
+import com.atlassian.bamboo.process.ProcessService;
+import com.atlassian.bamboo.task.TaskContext;
+import com.atlassian.bamboo.utils.SystemProperty;
+import com.atlassian.bamboo.v2.build.agent.capability.CapabilityContext;
+import com.atlassian.utils.process.ExternalProcess;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import net.lingala.zip4j.ZipFile;
+
+public class MatlabCommandRunner implements MatlabBuild {
+    private File tempDirectory;
+    private final CapabilityContext capabilityContext;
+    private final ProcessService processService;
+
+    public MatlabCommandRunner(ProcessService processService, CapabilityContext capabilityContext) {
+        this.capabilityContext = capabilityContext;
+        this.processService = processService;
+    }
+
+    public ExternalProcess run(String matlabCommand, TaskContext taskContext) throws IOException {
+        tempDirectory = getTempWorkingDirectory();
+        BuildLogger buildLogger = taskContext.getBuildLogger();
+        File workingDirectory = taskContext.getRootDirectory();
+        String matlabRoot = getMatlabRoot(taskContext, capabilityContext, buildLogger);
+        try {
+            buildLogger.addBuildLogEntry("Running MATLAB command: " + matlabCommand);
+            List<String> command = generateCommand(matlabCommand, workingDirectory);
+            ExternalProcessBuilder processBuilder = new ExternalProcessBuilder()
+                .workingDirectory(workingDirectory)
+                .command(command)
+                .env(SystemProperty.PATH.getKey(), matlabRoot);
+            ExternalProcess process = processService.createExternalProcess(taskContext, processBuilder);
+            process.execute();
+            return process;    
+        } finally {
+            clearTempDirectory(tempDirectory, buildLogger);
+            tempDirectory = null;
+        }
+    }
+
+    public void unzipToTempDir(String zipName) throws IOException {
```

Comment:
Seems like a single responsibility violation here that a command runner has a method to unzip to temp dir on it...

Reply:
Yeah... you caught me where I got lazy. All the temp folder logic was also in the `MatlabBuild` interface. I moved this one over because it was originally creating the test script as well, so I needed to change it anyways.

I can fix this by decoupling things a bit more sort of like how we have separate `script` and `matlab` modules in our Github actions. At that point, I think we should just delete that `MatlabBuild` interface all together.

Reply:
Which brings up the question... This interface is also in the Jenkins plugin. Should we consider a similar refactor there at some point?

Reply:
We can consider that in a separate dedicated pull. How about we just keep this one focused on the build tool integration?